spec/integration_spec.rb

```diff
@@ -58,4 +61,57 @@
       }.to raise_error(OpenSSL::SSL::SSLError, /certificate verify failed/)
     end
   end
+  
+  shared_examples "send payload" do
+    it "supports single element" do
+      (1..random_number_of_events).each do |n|
+        expect(client.write(payload)).to eq(sequence_start + n)
+      end
+      sleep(0.5) # give time to the server to read the events
+      expect(queue.size).to eq(random_number_of_events)
+    end
+
+    it "support sending multiple elements in one payload" do
+      expect(client.write(batch_payload)).to eq(sequence_start + batch_size)
+      sleep(0.5)
+
+      expect(queue.size).to eq(batch_size)
+      expect(queue).to match_array(batch_payload)
+    end
+  end
+
+  context "When transmitting a payload" do
+    let(:random_number_of_events) { Flores::Random.integer(2..10) }
+    let(:payload) { { "line" => "foobar" } }
+    let(:client) do
+      Lumberjack::Client.new(:port => port, 
+                             :host => host,
+                             :addresses => host,
+                             :ssl_certificate => certificate_file_crt)
+    end
+    let(:batch_size) { Flores::Random.integer(1..1024) }
+    let(:batch_payload) do
+      batch = []
+      batch_size.times do |n|
+        batch <<  { "line" => "foobar #{n}" }
+      end
+      batch
+    end
+
+    context "when sequence start at 0" do
+      let(:sequence_start) { 0 }
+
+      include_examples "send payload"
+    end
+
+    context "when sequence doesn't start at zero" do
+      let(:sequence_start) { Flores::Random.integer(1..2000) }
+
+      before do
+        client.instance_variable_get(:@socket).instance_variable_set(:@sequence, sequence_start)
```

Comment:
Not sure of the best way to change the sequence values without doing a bit of refactor.